src/simulator/shared.ts

```diff
@@ -1,11 +1,30 @@
 import { IPosition } from '../common';
-import { ID3SimulatorEngineSettings, ID3SimulatorEngineSettingsUpdate } from './engine/d3-simulator-engine';
 import { SimulationLinkDatum, SimulationNodeDatum } from 'd3-force';
+import { ID3SimulatorEngineSettings, ID3SimulatorEngineSettingsUpdate } from './engine/d3-simulator-engine';
+import { IEmitter } from '../utils/emitter.utils';
 
 export type ISimulationNode = SimulationNodeDatum & { id: number; mass?: number };
 export type ISimulationEdge = SimulationLinkDatum<ISimulationNode> & { id: number };
 
-export interface ISimulator {
+export enum SimulatorEventType {
+  STABILIZATION_STARTED = 'stabilizationStarted',
+  STABILIZATION_PROGRESS = 'stabilizationProgress',
+  STABILIZATION_ENDED = 'stabilizationEnded',
+  NODE_DRAGGED = 'nodeDragged',
+  NODE_DRAG_ENDED = 'nodeDragEnded',
+  SETTINGS_UPDATED = 'settingsUpdated',
```

Comment:
Nitpicking, but now when you cleaned up Events in simulator engines and main classes, maybe it makes sense to make naming consistent with the rest of the library where:

* Event values are present simple (e.g. `render-start`, `node-hover`, `mouse-move`)
* Event values are not camel case (e.g. `render-start`, ...)

Reply:
Yep! Makes sense. I'll also refactor these messages in the screenshot, as they are not written in `MACRO_CASE`.
![Screenshot from 2022-10-07 12-59-49](https://user-images.githubusercontent.com/3769376/194538613-7f7e93b2-0139-41cd-b146-573af1cfded5.png)

or maybe the best example is this one where both are clearly visible:
![Screenshot from 2022-10-07 13-03-15](https://user-images.githubusercontent.com/3769376/194538934-8b8324bc-14ea-4164-84b2-88ec103b8322.png)

Btw regarding naming, I have a few more questions:
* We are using both `stabilization` and `simulation`. Should we merge those two?
* In the picture above, should all of them be present simple?